Singly linked list using unique key code in cpp

anshmishra010 opened the pull request:
### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#pull-requests)?

Yes

### Description

Implemented Singly linked list using unique key code in cpp  #3393

### Checklist

- [x] I've read the contribution guidelines.
- [x] I've checked the issue list before deciding what to submit.
- [x] I've edited the `README.md` and link to my code.

## Related Issues or Pull Requests
added SLL Operation Using Unique Key in c-plus-plus/ds #3393 and updated readme.md

anshmishra010 committed:
added code for issue #3393

anshmishra010 committed:
added code for issue #3393

HarshCasper reviewed (changes requested):
Few changes: 
- Add a beginner-friendly description about your Code at the top
- Add description of functions 
- Add sample I/O at the end of the Code
- Add complexity at the end of the code
- Use a Code formatter to format the Code

anshmishra010 committed:
description, space and time complexity ,test cases added

anshmishra010 commented:
- [x] Added a beginner-friendly description about your Code at the top
- [x] Added description of functions
- [x] Added sample I/O at the end of the Code
- [x] Added complexity at the end of the code
- [x] Formatted the Code

anshmishra010 commented:
Hello harsh ,
I have updated the code , please check and reply .

On Tue, Mar 16, 2021, 11:29 PM Harsh Mishra ***@***.***>
wrote:

> *@HarshCasper* requested changes on this pull request.
>
> Few changes:
>
>    - Add a beginner-friendly description about your Code at the top
>    - Add description of functions
>    - Add sample I/O at the end of the Code
>    - Add complexity at the end of the code
>    - Use a Code formatter to format the Code
>
> —
> You are receiving this because you modified the open/close state.
> Reply to this email directly, view it on GitHub
> <https://github.com/TesseractCoding/NeoAlgo/pull/3403#pullrequestreview-613564886>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AQMPZXLNPV3RO7A6SI7KWLTTD6MBFANCNFSM4ZIOUSMA>
> .
>

HarshCasper commented on C-Plus-Plus/ds/SLL_Operations_Usnig_Unique_Key.cpp:
```diff
@@ -0,0 +1,314 @@
+/*
+This is singly linked list implementation in which a unique key is alloted while entering the new node to the linked list, and you cannot give same key to the different node. In this code you can give duplicate data still it work fine.
```
Modularize this and make the lines shorter and more readable

anshmishra010 replied:
- [x] Modularized the description and made the lines shorter and more readable.

HarshCasper commented on C-Plus-Plus/ds/SLL_Operations_Usnig_Unique_Key.cpp:
```diff
@@ -0,0 +1,314 @@
+Time complexity - O(n)
+
+Space complexity - O(n)
+*/
```
Add a new line at the EOF

anshmishra010 replied:
Added extra lines at the EOF

ankitaggarwal23 reviewed (changes requested):
@anshmishra010 Kindly do the changes above mentioned.

anshmishra010 committed:
Merge branch 'master' of https://github.com/TesseractCoding/NeoAlgo

anshmishra010 committed:
description changed and added an empty line at EOF
